### PR TITLE
std.math: Move most of the rounding functions into submodule.

### PR DESCRIPTION
**Pull request by berni44**
I left out `floor` and `ceil` because moving `floorImpl` to the submodule fails with a strange error message. I guess, there is some compiler magic involved here, making `union` work in CTFE.

### CODE REVIEW

**Comment by berni44** [comment]
Here `compilable/test5227.d` fails. It's two function that cannot be evaluated at runtime: `round` and `trunc`. Looks, like the calls to `core.stdc.math.roundl` and `core.stdc.math.truncl` in CTFE are restricted to `std.math`, but cannot be called from `std.math.rounding`.

**Comment by berni44** [comment]
It's not the calls I mentioned above. So I suspect dmd searching `round` and `trunc` in `std.math` and not in `std.math.rounding`; that is, the public imports are ignored... I need help here from someone who knows the inner workings of dmd.

**Comment by thewilsonator** [comment]
https://github.com/dlang/dmd/blob/master/src/dmd/builtin.d#L91 is the function in question.

**Comment by RazvanN7** [comment]
Have you tried making a public alias to the symbols that were moved to the submodules? That should work.

Otherwise, we can make the compiler acknowledge the new submodules, although I would rather not go down that route.

**Comment by berni44** [comment]
> Have you tried making a public alias to the symbols that were moved to the submodules? That should work.

Wouldn't that clash with the public imports? I'll check it out.

**Comment by berni44** [comment]
Didn't clash, so I probably do not understand something with visibility and scope... But that's my fault. I'll try to pick that up.

Anyway, the error has not vanished, but this time brought up a new detail:

```
D:\a\1\dmd\test\..\..\phobos\std\math\package.d(4851): Error: `asm` statements cannot be interpreted at compile time
D:\a\1\dmd\test\..\..\phobos\std\math\rounding.d(437):        called from here: `getControlState()`
compilable\test5227.d(70):        called from here: `round(5.5L)`
compilable\test5227.d(70):        while evaluating `pragma(msg, roundr)`
```

Line 4851 of `package.d` is inside `getControlState()` of `FloatingPointControl`.

File: std/math/rounding.d [hunk 1]
```diff
@@ -19,4 +19,573 @@ Source: $(PHOBOSSRC std/math/rounding.d)
...
+ *
+ * This is generally the fastest method to convert a floating-point number
+ * to an integer. Note that the results from this function
+ * depend on the rounding mode, if the fractional part of x is exactly 0.5.
+ * If using the default rounding mode (ties round to even integers)
+ * lrint(4.5) == 4, lrint(5.5)==6.
+ */
+long lrint(real x) @trusted pure nothrow @nogc
+{
+    version (InlineAsm_X87)
```
Comment (ibuclaw) [inline std/math/rounding.d:223]: Missing version definition at top of file.

File: std/math/rounding.d [hunk 2]
```diff
@@ -19,4 +19,573 @@ Source: $(PHOBOSSRC std/math/rounding.d)
...
+}
+
+/**
+ Returns the integer portion of x, dropping the fractional portion.
+ This is also known as "chop" rounding.
+ `pure` on all platforms.
+ */
+real trunc(real x) @trusted nothrow @nogc pure
+{
+    version (InlineAsm_X87_MSVC)
```
Comment (ibuclaw) [inline std/math/rounding.d:504]: Missing version definition

File: std/math/package.d [hunk 3]
```diff
@@ -167,6 +174,9 @@ deprecated("std.meta.AliasSeq was unintentionally available from std.math "
            ~ "and will be removed after 2.102. Please import std.meta instead")
 public import std.meta : AliasSeq;
 
+public alias round = std.math.rounding.round;
+public alias trunc = std.math.rounding.trunc;
```
Comment (ibuclaw) [inline std/math/package.d:178]: Remove this as is unnecessary

File: std/math/rounding.d [hunk 4]
```diff
@@ -19,4 +19,573 @@ Source: $(PHOBOSSRC std/math/rounding.d)
 
 module std.math.rounding;
 
-// Will contain functions like ceil, floor, round, quantize
+static import core.math;
+static import core.stdc.math;
+
+import std.traits : isFloatingPoint, isIntegral, Unqual;
+
```
Comment (ibuclaw) [inline std/math/rounding.d:26]:
```suggestion

version (D_InlineAsm_X86)    version = InlineAsm_X86_Any;
version (D_InlineAsm_X86_64) version = InlineAsm_X86_Any;

version (InlineAsm_X86_Any) version = InlineAsm_X87;
version (InlineAsm_X87)
{
    static assert(real.mant_dig == 64);
    version (CRuntime_Microsoft) version = InlineAsm_X87_MSVC;
}

```

**Review by ibuclaw — CHANGES_REQUESTED** [review]
Squash and add missing versions.

**Review by ibuclaw — APPROVED** [review]

**Pull request merged**